src/main/java/com/redhat/cloud/notifications/routers/UserConfigService.java

```diff
@@ -147,48 +149,57 @@
      * Pulls the user settings values of an user across all the know applications of a bundle
      * */
     private Uni<SettingsValues> getSettingsValueForUser(String account, String username, String bundleName) {
-        return Uni.createFrom().deferred(() -> {
-            final SettingsValues values = new SettingsValues();
-
-            if (bundleName == null || bundleName.equals("")) {
-                throw new BadRequestException("bundleName must have a value");
-            }
-
-            return bundleResources.getBundle(bundleName)
-                    .onItem().ifNull().failWith(() -> new BadRequestException("Unknown bundleName: " + bundleName))
-                    .onItem().transformToMulti(bundle -> {
-                        BundleSettingsValue bundleSettingsValue = new BundleSettingsValue();
-                        bundleSettingsValue.displayName = bundle.getDisplayName();
-                        values.bundles.put(bundle.getName(), bundleSettingsValue);
-                        return applicationResources.getApplications(bundle.getName())
-                                .onItem().transformToMulti(application -> {
-                                    ApplicationSettingsValue applicationSettingsValue = new ApplicationSettingsValue();
-                                    applicationSettingsValue.displayName = application.getDisplayName();
-                                    EmailTemplate applicationEmailTemplate = emailTemplateFactory.get(bundle.getName(), application.getName());
-
-                                    values.bundles.get(bundle.getName()).applications.put(application.getName(), applicationSettingsValue);
-                                    return Multi.createFrom().items(EmailSubscriptionType.values())
-                                            .onItem().transformToMulti(emailSubscriptionType -> {
-                                                if (applicationEmailTemplate.isEmailSubscriptionSupported(emailSubscriptionType)) {
-                                                    values.bundles.get(bundle.getName()).applications.get(application.getName()).notifications.put(emailSubscriptionType, false);
+        if (bundleName == null || bundleName.equals("")) {
+            throw new BadRequestException("bundleName must have a value");
+        }
+
+        /*
+         * The following stream creates a `SettingsValues` instance and then populates it by using side-effects.
+         * It is done by invoking to kind of functions:
+         * - database queries are executed in an asynchronous way using `.onItem().call(...)`
+         * - side-effects are executed in a synchronous way using `.onItem().invoke(...)`
+         */
+        return Uni.createFrom().item(() -> new SettingsValues())
```

Comment:
This is roughly the same code than before but we're no longer using `Multi` in this method.